axelrod/tests/strategies/test_grudger.py

```diff
@@ -44,34 +51,36 @@ class TestForgetfulGrudger(TestPlayer):
     }
 
     def test_strategy(self):
-        self.responses_test([C], attrs={"grudged": False})
-        self.responses_test([C], [C], [C], attrs={"grudged": False})
-        self.responses_test([D], [C], [D], attrs={"grudged": True})
-        for i in range(10):
-            self.responses_test([D], [C, C] + [D] * i, [C, D] + [C] * i,
-                                attrs={"grudged": True, "grudge_memory": i,
-                                       "mem_length": 10})
-        # Forgets the grudge eventually
-        i = 10
-        self.responses_test([C], [C, C] + [D] * i + [C], [C, D] + [C] * i + [C],
-                            attrs={"grudged": False, "grudge_memory": 0,
-                                   "mem_length": 10})
+        # If opponent defects at any point then the player will respond with
+        # D ten times and then continue to check for defections.
+        opponent = axl.Cooperator()
+        actions = [(C, C)] * 20
+        self.versus_test(opponent, expected_actions=actions)
```

Comment:
These look great but we're losing the `attrs` test above. Could you add them in?

From: http://axelrod.readthedocs.io/en/latest/tutorials/contributing/strategy/writing_test_for_the_new_strategy.html:

> The function versus_test also accepts a dictionary parameter of attributes to check at the end of the match. For example this test checks if the player’s internal variable opponent_class is set to "Cooperative":

Let me know if I can assist.